homeassistant/components/binary_sensor/zha.py

```diff
@@ -87,3 +124,102 @@ def cluster_command(self, tsn, command_id, args):
         elif command_id == 1:
             _LOGGER.debug("Enroll requested")
             self.hass.add_job(self._ias_zone_cluster.enroll_response(0, 0))
+
+
+class Switch(zha.Entity, BinarySensorDevice):
+    """ZHA switch/remote controller/button."""
+
+    _domain = DOMAIN
+
+    class OnOffListener:
+        """Listener for the OnOff ZigBee cluster."""
+
+        def __init__(self, entity):
+            """Initialize OnOffListener."""
+            self._entity = entity
+
+        def cluster_command(self, tsn, command_id, args):
+            """Handle commands received to this cluster."""
+            if command_id in (0x0000, 0x0040):
+                self._entity._set_state(False)
+            elif command_id in (0x0001, 0x0041, 0x0042):
+                self._entity._set_state(True)
+            elif command_id == 0x0002:
+                self._entity._set_state(not self._entity._state)
+
+        def attribute_updated(self, attrid, value):
+            """Handle attribute updates on this cluster."""
+            if attrid == 0:
+                self._entity._state = value
+                self._entity.schedule_update_ha_state()
+
+    class LevelListener:
+        """Listener for the LevelControl ZigBee cluster."""
+
+        def __init__(self, entity):
+            """Initialize LevelListener."""
+            self._entity = entity
+
+        def cluster_command(self, tsn, command_id, args):
+            """Handle commands received to this cluster."""
+            if command_id == 0x0001:  # move_to_level
+                self._entity._set_level(args[0])
+            elif command_id == 0x0002:  # step
+                # Step (technically shouldn't change on/off)
+                if args[0] == 0:
+                    self._entity._move_level(args[1])
+                else:
+                    self._entity._move_level(-args[1])
+            elif command_id == 0x0004:  # move_to_level_with_on_off
+                self._entity._set_level(args[0])
```

Comment:
should we update the 'state' since this is a "with_on_off" command?

Reply:
Never mind. I see now it's being handled in the _move_level. Although per ZCL specs, move, move_to_level, step, stop commands shall be ignored when the device is in 'Off' state.